subsys/bluetooth/mesh/access.c

```diff
@@ -1810,15 +1996,21 @@ static void store_pending_mod_pub(struct bt_mesh_model *mod, bool vnd)
 	if (!mod->pub || mod->pub->addr == BT_MESH_ADDR_UNASSIGNED) {
 		err = settings_delete(path);
 	} else {
-		pub.addr = mod->pub->addr;
-		pub.key = mod->pub->key;
-		pub.ttl = mod->pub->ttl;
-		pub.retransmit = mod->pub->retransmit;
-		pub.period = mod->pub->period;
-		pub.period_div = mod->pub->period_div;
-		pub.cred = mod->pub->cred;
+		pub.base.addr = mod->pub->addr;
+		pub.base.key = mod->pub->key;
+		pub.base.ttl = mod->pub->ttl;
+		pub.base.retransmit = mod->pub->retransmit;
+		pub.base.period = mod->pub->period;
+		pub.base.period_div = mod->pub->period_div;
+		pub.base.cred = mod->pub->cred;
+
+		if (BT_MESH_ADDR_IS_VIRTUAL(mod->pub->addr)) {
+			(void)bt_mesh_va_get_idx_by_uuid(mod->pub->uuid, &pub.uuidx);
+		}
 
-		err = settings_save_one(path, &pub, sizeof(pub));
+		err = settings_save_one(path, &pub,
```

Comment:
`base` substructure came from backward compatibility. Even if I remove this and store the whole structure, the code for reading it will stay as is to keep backward compatibility. I can swap around line 1744 and 1742 so that `base` version is always read first as most likely to happen.